src/libpmemkv.cc

```diff
@@ -250,104 +250,112 @@ pmemkv_db *pmemkv_open(void *context, const char *engine_c_str, pmemkv_config *c
 
 #ifdef ENGINE_VSMAP
 		if (engine == "vsmap") {
-			return reinterpret_cast<pmemkv_db *>(
+			*db = reinterpret_cast<pmemkv_db *>(
 				new pmem::kv::vsmap(context, path.get(), size));
 		}
 #endif
 
 #ifdef ENGINE_VCMAP
 		if (engine == "vcmap") {
-			return reinterpret_cast<pmemkv_db *>(
+			*db = reinterpret_cast<pmemkv_db *>(
 				new pmem::kv::vcmap(context, path.get(), size));
 		}
 #endif
 		throw std::runtime_error("Unknown engine name");
 	} catch (std::exception &e) {
-		onfail(context, engine_c_str, config, e.what());
-		return nullptr;
+		std::cerr << e.what() << "\n";
+		*db = nullptr;
+
+		return PMEMKV_STATUS_FAILED;
 	}
+
+	return PMEMKV_STATUS_OK;
```

Comment:
this should be returned earlier I believe, since e.g. blackhole and caching will fail somewhere around get_config (line 211)

Reply:
Done.